docs/iam-usage.rst

```diff
@@ -0,0 +1,177 @@
+
+Policies
+~~~~~~~~
+
+A policy is a dictionary from role strings to sets of members. ``resource.get_policy()`` also provides etags and versions. An etag is used to provide optimistic concurrency controls on policy updates, while versions are provided for end-user versioning.
+
+
+Policy Changes
+~~~~~~~~~~~~~~
+
+An ``iam.PolicyChange`` object encapsulates changes to made to a policy transactionally. The ``iam.PolicyChange`` constructor takes an optional ``version`` keyword argument, an integer to use as the policy version. If ``version`` is ``None``, when applied ``iam.PolicyChange`` will increment whatever the current version of the policy is by 1.
+
+To apply a ``iam.PolicyChange`` to a resource which implements the IAM interface call the ``apply()`` method with the resource as an argument. 
+
+>>> policy_change = iam.PolicyChange(version=2).add(iam.roles.OWNER.name, [iam.user('alice@example.com')])
+>>> policy, version, etag = policy_change.apply(resource)
+>>> print(policy)
+{'roles/owner': set(['user:alice@example.com'])}
+>>> print(version)
+2
+>>> print(etag)
+xDSFbfdasfAEFdfCds
+
+Apply returns the new policy, it's version, and it's etag.
+
+Optionally, a ``version`` keyword argument can be supplied to ``apply`` which will override the ``version`` behavior of the policy change.
+
+>>> _, version, _ = policy_change.apply(resource, version=None)
+>>> print(version)
+3
+
+Modifications can be added to a ``iam.PolicyChange`` object by one of two methods:
+
+First the user can directly add or remove members from a ``Role``. ``iam.PolicyChange`` exposes two methods for this, ``add`` and ``remove`` which both take a role string or ``iam.Role`` object and a list of member strings.
+
+>>> policy_change = iam.PolicyChange().add(iam.roles.OWNER, [iam.user('alice@example.com')])
+>>> policy_change.remove(iam.roles.EDITOR, [iam.domain('example.com'), iam.group('devs@example.com')])
+>>> policy, _, _ = policy_change.apply(resource)
+>>> print(iam.user('alice@example.com') in policy[iam.roles.OWNER.name])
+True
+>>> print(iam.domain('example.com') in policy[iam.roles.EDITOR.name])
+False
+
+Second the user can specify a "membership function" which will take a member string as an argument, and return ``True`` if the member should belong to the specified role, and ``False`` otherwise.
+
+>>> def membership_fn(member):
+>>>     return not iam.is_group(member) or member == iam.user('bob@example.com')
+>>> policy_change.fn(iam.roles.READER, membership_fn)
+>>> policy, _, _ policy_change.apply(resource)
+>>> print([member for member in policy[iam.roles.READER.name] if iam.is_group(member)])
+['user:bob@example.com']
+
+
+Methods
+----------------------------------
+
+Resources that implement the IAM interface provide the following methods:
+
+Low Level Methods
+~~~~~~~~~~~~~~~~~
+Resources that implement IAM provide low level methods for interacting with IAM. 
+
+``get_policy`` returns a tuple of ``(policy, version, etag)`` on the corresponding resource. 
```

Comment:
nit: "triple of `policy, version, etag`" would work just as well (i.e. the parentheses aren't needed). But feel free to leave as-is.